src/csharp/Grpc.Tools/ProtoCompile.cs

```diff
@@ -136,7 +136,7 @@ public class ProtoCompile : ToolTask
             new ErrorListFilter
             {
                 Pattern = new Regex(
-                    pattern: "(?'FILENAME'.+)\\((?'LINE'\\d+)\\) ?: ?warning in column=(?'COLUMN'\\d+) ?: ?(?'TEXT'.*)",
+                    pattern: "(?'FILENAME'[^\\(]+)\\((?'LINE'\\d+)\\) ?: ?warning in column=(?'COLUMN'\\d+) ?: ?(?'TEXT'.*)",
```

Comment:
I've had the same idea, but there's a small glitch that `(` character is legal in a file name. So I think for file names containing parentheses, this will break the error reporting. 
I think this is going to a rare cornercase, so it's worth merging anyway to fix the flakes, but it's not a 100% solution.
Perhaps we should add a TODO to the code at least?

Reply:
Ah. that's a fair point. I'm not too familiar with regex, so excuse my premature fix but I think there's a better solution. I'll update the PR description.